### PR TITLE
fix issue with MethodRenamer

### PR DESCRIPTION
**Pull request by gnom7**
when class already contains method-1 and field with the same name and we decide to rename another method-2 with same field's(method-1's) name, so getting for a while two different methods with same name (if method-2 renamed before method-1) in one class, causing Exception)

### CODE REVIEW

**Comment by gnom7** [comment]
@mbenz89 may I ask you to merge this pr please?

File: src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java [hunk 1]
```diff
@@ -214,51 +214,52 @@ public static String getNewName() {
...
+                .filter(SootClass::isLibraryClass)
+                .anyMatch(c -> c.declaresMethod(method.getName(), method.getParameterTypes(), method.getReturnType()));
+    }
+
+    private static List<SootClass> getAllInterfacesOf(SootClass sc) {
+        Stream<SootClass> superClassInterfaces = sc.isInterface() ? Stream.empty() : hierarchy.getSuperclassesOf(sc)
+                .stream()
+                .map(MethodRenamer::getAllInterfacesOf)
+                .flatMap(Collection::stream);
+        Stream<SootClass> directInterfaces = Stream.concat(sc.getInterfaces().stream(), sc.getInterfaces().stream()
```
Comment (mbenz89) [inline src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java]: You concat the same stream here, is this correct? If so, could you please explain the semantics behind it?

File: src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java [hunk 2]
```diff
@@ -85,81 +85,80 @@ protected void internalTransform(String phaseName,
...
-									+ c.getName());
+									+ sc.getName());
+						}
 						rename = false;
 						break;
 					}
 				}
 
 				if (rename) {
-					// TODO: This is flawed since it all methods of a similar
```
Comment (mbenz89) [inline src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java]: Where you able to fix t this? Where?

File: src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java (hunk 1 above)
Comment (gnom7) [reply src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java]: Please re-check this, second stream is not the same as first.

File: src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java (hunk 2 above)
Comment (gnom7) [reply src/main/java/soot/jbco/jimpleTransformations/MethodRenamer.java]: I will provide commit for this in a while (after I check changes on my project and ensure that all changes work fine), for now I have returned this TODO in f31aba60e76f438973c61abcf85d78b2e33d5392

**Comment by mbenz89** [comment]
OK to test

**Comment by mbenz89** [comment]
Thanks for your contribution!

**Pull request merged**